lib/mini_defender/rules/url.rb

```diff
@@ -1,17 +1,79 @@
 # frozen_string_literal: true
 
 require 'uri'
+require 'resolv'
+require_relative '../detectors/public_domain_detector'
 
 class MiniDefender::Rules::Url < MiniDefender::Rule
+  ALLOWED_MODIFIERS = %w[https_only public no_ip]
+
+  def initialize(modifiers = [])
+    @modifiers = Array(modifiers).map(&:to_s)
+    validate_modifiers! unless @modifiers.empty?
+  end
+
   def self.signature
     'url'
   end
 
+  def self.make(args)
+    new(args)
+  end
+
   def passes?(attribute, value, validator)
-    value.is_a?(String) && URI.regexp(%w[http https]).match?(value)
+    # TODO: warning: URI.regexp is obsolete; use URI::DEFAULT_PARSER.make_regexp instead
+    return false unless value.is_a?(String) && URI.regexp(%w[http https]).match?(value)
+
+    begin
+      uri = URI.parse(value)
+
+      return true if @modifiers.empty?
```

Comment:
Use `if` statement instead of `if` modifier

Reply:
Noted 🌟